compiler/rustc_lint/src/unused.rs

```diff
@@ -1349,9 +1349,9 @@ declare_lint! {
     /// ### Example
     ///
     /// ```rust
-    /// #![feature(box_syntax)]
+    /// #![feature(rustc_attrs, stmt_expr_attributes)]
     /// fn main() {
-    ///     let a = (box [1, 2, 3]).len();
+    ///     let a = (#[rustc_box] Box::new([1, 2, 3])).len();
```

Comment:
When #104363 also makes it lint for `Box::new` this should be updated, I'll leave a comment there

Reply:
i did indeed forget to leave a comment there, great
i will do it now